### PR TITLE
feat: Handle permission driven views for auto-saving pages and action…

### LINKED ISSUE #13014: [Task]: Permission driven views - 403 forbidden error
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

- [x] #16458 
- [x] #16459 
- [x] #16460 
- [x] #16461

### PR DESCRIPTION
**Pull request by ankitakinger**
## Description

> Adding permission checks on granular level to support Granular access control on enterprise edition.

Fixes #13014 #16443 

## Type of change

- New feature (non-breaking change which adds functionality)
- Refactor

## How Has This Been Tested?

> Testing this via all integration tests and manual testing 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/ce/utils/adminSettingsHelpers.ts [hunk 1]
```diff
@@ -29,3 +33,15 @@ export const saveAllowed = (settings: any) => {
...
 };
+
+/* get default admin settings path */
+export const getDefaultAdminSettingsPath = (
+  isSuperUser = false,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  tenantPermissions?: string[],
+): string => {
+  if (isSuperUser) {
+    return ADMIN_SETTINGS_CATEGORY_DEFAULT_PATH;
```
Comment (ankitakinger) [inline app/client/src/ce/utils/adminSettingsHelpers.ts:44]: needs to be updated

**Review by berzerkeer — APPROVED** [review]

**Pull request merged**